src/Filesystem/Folder.php

```diff
@@ -416,21 +416,26 @@ public function inCakePath($path = '')
     }
 
     /**
-     * Returns true if the File is in given path.
+     * Returns true if the Folder is in the given path.
      *
-     * @param string $path The path to check that the current pwd() resides with in.
-     * @param bool $reverse Reverse the search, check that pwd() resides within $path.
+     * @param string $path The absolute path to check that the current `pwd()` resides within. If omitted
+     *  (or empty), the current top level directory is assumed.
+     * @param bool $reverse Reverse the search, check if the given `$path` resides within the current `pwd()`.
      * @return bool
      */
     public function inPath($path = '', $reverse = false)
     {
+        if (empty($path)) {
```

Comment:
Ok... and what are your thoughts on throwing an exception for non-absolute paths, and removing the "let's assume the top level directory for empty paths" behavior altogether (having in mind that it wasn't documented, always triggered a warning, and never worked on Windows)?

If that's too hard of a break, it could surely be done a little more gracefully, and step by step.